### PR TITLE
Add new Sample Format guidelines. 

### PR DESCRIPTION
**Pull request by kurtisvg**
Add a document summarizing some java-specific requirements for the new sample format.

### CODE REVIEW

**Comment by lesv** [comment]
You should update CONTRIBUTING.md as well

**Comment by kurtisvg** [comment]
> You should update CONTRIBUTING.md as well

My plan is to wait until a decision is made on when the sample format becomes required before updating that.

File: SAMPLE_FORMAT.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ exceptions can be handled by logging the exception to `System.out.println`. If the exception is 
+ something the user may commonly encounter, include a comment explaining how to avoid or handle 
+ correctly.
+
+Example:
+```java
+try {
+  // Do something
+} catch (Exception e) {
+  System.out.println("Error during functionName: \n" + e.toString());
```
Comment (lesv) [inline SAMPLE_FORMAT.md]: It would be better to show a common exception and good comment here.

File: SAMPLE_FORMAT.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+## Additional Best Practices 
+
+The following are some general Java best practices that should be followed in samples to remain
+idiomatic. 
+
+### Time
+Use the `java.time` package when dealing with units of time in some manner. 
+
+### Logging
+Use `java.util.logging` for consistent logging in web applications.
```
Comment (lesv) [inline SAMPLE_FORMAT.md]: Should we move to https://github.com/google/flogger at some point?

Comment (kurtisvg) [reply SAMPLE_FORMAT.md]: Added to todo list for discussion.

**Pull request merged**